libs/shared/ui-components/src/Video/Video.tsx

```diff
@@ -0,0 +1,131 @@
+
+const playerOptions = {
+  autoPlay: 0,
+  disablekb: 1,
+  showinfo: 1,
+  volume: 0,
+  muted: 0,
+  pip: 1,
+  controls: 2,
+  enablejsapi: 1,
+};
+
+const videoSizeStyles: Record<TVideoProps['size'], string> = {
+  small: 'lg:w-1/2 lg:min-w-[50%] lg:pt-[28.125%]',
+  medium: 'lg:w-3/4 lg:min-w-[75%] lg:pt-[42.18%]',
+  large: '',
+};
+
+const alignVideoStyles: Record<TVideoProps['align'], string> = {
+  left: 'lg:flex-row',
+  right: 'lg:flex-row-reverse',
+  center: '',
+};
+
+const topSpaceStyles: Record<TVideoProps['topSpace'], string> = {
+  none: 'mt-0',
+  small: 'mt-[4rem] xl:mt-[8rem]',
+  large: 'mt-[6rem] xl:mt-[10rem]',
+};
+
+const bottomSpaceStyles: Record<TVideoProps['bottomSpace'], string> = {
+  none: 'mb-0',
+  small: 'mb-[4rem] xl:mb-[8rem]',
+  large: 'mb-[6rem] xl:mb-[10rem]',
+};
+
+const textStyles =
+  'my-[1.6rem] text-[1.6rem] text-black marker:text-black prose sm:text-[1.8rem] sm:my-[4rem] xl:leading-[2.7rem]';
+
+export const Video = ({
+  url,
+  thumbnail,
+  size,
+  align,
+  title,
+  sideText,
+  text,
+  topSpace,
+  bottomSpace,
+}: TVideoProps) => {
+  const [playing, setPlaying] = useState(false);
+
+  const handleToggleVideoPlay = () => setPlaying((prevState) => !prevState);
+
+  if (!url) {
+    return null;
+  }
+
+  const playerConfig = {
+    vimeo: { playerOptions },
+    youtube: { playerVars: playerOptions },
+  };
+
+  const styles = {
+    header: `
+      mb-[5.6rem] text-[4rem] font-extrabold leading-[4.9rem] text-center font-heading text-violet
+      sm:mx-auto sm:max-w-[70rem] sm:text-[4.8rem] sm:leading-[5.3rem]
+      xl:mb-[6.8rem]
+    `,
+    player: clsx(
+      'relative pt-[56.25%] w-full',
+      videoSizeStyles[size || 'large'],
+    ),
+    text: clsx(textStyles, 'hidden', size === 'small' && 'lg:block'),
+    video: clsx(
+      size === 'small' && 'lg:flex lg:gap-[4rem] lg:items-center xl:gap-[8rem]',
+      size === 'small' && alignVideoStyles[align || 'left'],
+    ),
+    wrapper: clsx(
+      'px-[1.8rem] mx-auto md:px-[3rem] xl:px-[15rem] max-w-layout',
+      topSpaceStyles[topSpace || 'small'],
+      bottomSpaceStyles[bottomSpace || 'small'],
+    ),
+  };
+
+  return (
+    <div className={styles.wrapper}>
+      {title && <h2 className={styles.header}>{title}</h2>}
+      <div className={styles.video}>
+        <div className={styles.player}>
+          {thumbnail?.filename && (
+            <VideoThumbnail
+              src={thumbnail.filename}
+              alt={thumbnail.alt || 'video thumbnail'}
+              onClick={handleToggleVideoPlay}
+            />
+          )}
+          <ReactPlayer
+            width="100%"
+            height="100%"
+            className="absolute top-0 left-0"
+            url={url}
+            playing={playing}
+            config={playerConfig}
+          />
+        </div>
+        {sideText && (
+          <div
+            className={styles.text}
+            dangerouslySetInnerHTML={createMarkup(sideText)}
```

Comment:
Just double checking: these values for `dangerouslySetInnerHTML` are only coming from Storyblok, right?

Reply:
Yes, that's the way of implementing richText fields.